core/shared/src/main/scala/org/http4s/headers/Cookie.scala

```diff
@@ -28,23 +28,24 @@ object Cookie {
   def apply(head: RequestCookie, tail: RequestCookie*): `Cookie` =
     apply(NonEmptyList(head, tail.toList))
 
-  def parse(s: String): ParseResult[Cookie] =
-    ParseResult.fromParser(parser, "Invalid Cookie header")(s)
+  def parse(s: String): ParseResult[Cookie] = {
+    val (errors, cookies) = s
+      .split("; |;")
+      .toList
+      .filterNot(_.isEmpty)
+      .map(parser.parseAll)
```

Comment:
Invoking the parser multiple times is going to hurt performance a bit, but our parser isn't currently built to return an Ior in one pass.  I think we could fix this later if it became a problem in practice.